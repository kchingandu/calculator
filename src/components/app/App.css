.App {
	text-align: center;
}

* {
	margin: 0;
	padding: 0;
}

html {
	font-family: sans-serif;
	font-size: 62.5%;
}

body {
	display: flex;
	height: 100vh;
	padding: 1.4rem;
	font-size: 1.4rem;
	background: rgb(249,220,212);
	background: linear-gradient(35deg, rgba(249,220,212,1) 0%, rgba(233,194,214,1) 48%, rgba(206,172,208,1) 100%);
}

hr {
	opacity: 0;
}

button {
	width: 5rem;
	border: none;
	height: 5rem;
	outline: none;
	color: #442939;
	cursor: pointer;
	font-weight: bold;
	font-size: 1.4rem;
	margin: 1rem .5rem;
	border-radius: 2.5rem;
	filter: drop-shadow(0px 0px 3px #dabdf1);
	transition: color 650ms, background 650ms;
	background: linear-gradient(0deg, rgba(242,210,222,1) 0%, rgba(255,233,234,1) 62%, rgba(255,252,248,1) 100%);
}

button:hover,
button:focus {
	color: white;
	background: #442939;
	transition: color 650ms, background 650ms;
}

#root {
	margin: auto;
}

#clear {
	height: 3rem;
	width: 23rem;
}

#info{
	display: block;
	font-size: 1.2rem;
	margin: 10rem 0 0 0;
}

#display {
	width: 100%;
	color: white;
	height: 4rem;
	border: none;
	outline: none;
	margin: 2rem 0;
	padding: 0 1rem;
	font-size: 1.6rem;
	text-align: end;
	user-select: none;
	background: #442939;
	border-radius: 1rem;
	letter-spacing: .2rem;
	box-sizing: border-box;
	filter: drop-shadow(0px 0px 3px #dabdf1);
}

#calculator {
	width: 24rem;
	padding: 2rem;
	margin: 0 auto;
	user-select: none;
	border-radius: 2rem;
	background: rgb(242,210,222);
	filter: drop-shadow(0px 0px 8px #b29ac4);
	background: linear-gradient(0deg, rgba(242,210,222,1) 0%, rgba(255,233,234,1) 62%, rgba(255,252,248,1) 100%);
}