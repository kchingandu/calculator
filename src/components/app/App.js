import React, { useState, useEffect } from 'react';
import './App.css';
import Calculator from '../caluclator/Cacluclator';
import { getClickAndKeyDown$ } from '../../support/streams/streams';

const App = () => {
  const [displayText, setDisplayText] = useState('0');

  useEffect(() => {
    const subscription = getClickAndKeyDown$().subscribe((result) => {
      setDisplayText(result);

      document.activeElement && document.activeElement.blur();
    });

    return () => subscription();
  }, []);

  return (
    <div className="App">
      <Calculator displayText={displayText}/>

      <span id="info">
        c key = clear | Enter key = answer | Backsapce key = delete | 0-9 + / - x * (=) = supported inputs | 1+2* = generates error
      </span>
    </div>
  );
};

export default App;
