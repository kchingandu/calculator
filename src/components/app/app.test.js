import App from './App';
import React from 'react';
import { mount } from 'enzyme';

describe('App', () => {
  let app;

  beforeEach(() => {
    app = mount(<App/>);
  });

  it('should render the calculator and info text', () => {
    expect(app.html()).toMatchSnapshot();
  });
});