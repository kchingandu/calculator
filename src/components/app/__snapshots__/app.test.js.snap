// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the calculator and info text 1`] = `"<div class=\\"App\\"><div id=\\"calculator\\"><input dir=\\"ltr\\" id=\\"display\\" value=\\"0\\"><button class=\\"calculator-button\\">7</button><button class=\\"calculator-button\\">8</button><button class=\\"calculator-button\\">9</button><button class=\\"calculator-button\\">รท</button><button class=\\"calculator-button\\">4</button><button class=\\"calculator-button\\">5</button><button class=\\"calculator-button\\">6</button><button class=\\"calculator-button\\">x</button><button class=\\"calculator-button\\">1</button><button class=\\"calculator-button\\">2</button><button class=\\"calculator-button\\">3</button><button class=\\"calculator-button\\">-</button><button class=\\"calculator-button\\">0</button><button class=\\"calculator-button\\">.</button><button class=\\"calculator-button\\">=</button><button class=\\"calculator-button\\">+</button><button id=\\"clear\\" type=\\"button\\" class=\\"calculator-button\\">c</button></div><span id=\\"info\\">c key = clear | Enter key = answer | Backsapce key = delete | 0-9 + / - x * (=) = supported inputs | 1+2* = generates error</span></div>"`;
